libcloud/compute/drivers/gce.py

```diff
@@ -2252,29 +2321,112 @@ def ex_create_route(self, name, dest_range, priority=500,
 
         return self.ex_get_route(name)
 
-    def ex_create_network(self, name, cidr, description=None):
+    def ex_create_subnetwork(self, name, cidr=None, network=None, region=None,
+                             description=None):
         """
-        Create a network.
+        Create a subnetwork.
 
-        :param  name: Name of network to be created
+        :param  name: Name of subnetwork to be created
         :type   name: ``str``
 
         :param  cidr: Address range of network in CIDR format.
         :type   cidr: ``str``
 
+        :param  network: The network name or object this subnet belongs to.
+        :type   network: ``str`` or :class:`GCENetwork`
+
+        :param  region: The region the subnetwork belongs to.
+        :type   region: ``str`` or :class:`GCERegion`
+
         :param  description: Custom description for the network.
         :type   description: ``str`` or ``None``
 
+        :return:  Subnetwork object
+        :rtype:   :class:`GCESubnetwork`
+        """
+        if not cidr:
+            raise ValueError("Must provide an IP network in CIDR notation.")
+
+        if not network:
+            raise ValueError("Must provide a network for the subnetwork.")
+        else:
+            if isinstance(network, GCENetwork):
+                network_url = network.extra['selfLink']
+            else:
+                if network.startswith('https://'):
+                    network_url = network
+                else:
+                    network_obj = self.ex_get_network(network)
+                    network_url = network_obj.extra['selfLink']
+
+        if not region:
+            raise ValueError("Must provide a region for the subnetwork.")
+        else:
+            if isinstance(region, GCERegion):
```

Comment:
Refactoring this region related functionality in a separate utility method would be nice.

Again not a blocker right now, but it would be great we at least have tests for all those various ways you can pass region in (if there aren't any already) so we can easily refactor it later on.